core/src/main/java/org/elasticsearch/index/search/child/ChildrenConstantScoreQuery.java

```diff
@@ -103,11 +103,8 @@ public Weight doCreateWeight(IndexSearcher searcher, boolean needsScores) throws
             return new BooleanQuery().createWeight(searcher, needsScores);
         }
 
-        IndexSearcher indexSearcher = new IndexSearcher(searcher.getIndexReader());
-        indexSearcher.setSimilarity(searcher.getSimilarity(true));
-        indexSearcher.setQueryCache(null);
         ParentOrdCollector collector = new ParentOrdCollector(globalIfd, valueCount, parentType);
-        indexSearcher.search(childQuery, collector);
+        sc.searcher().search(childQuery, collector);
```

Comment:
right now this can't be done, because in the case of a dfs_* search the provided searcher can be of type `CachedDfSource` and that implementation can't be used for searching.

Maybe we should merge `CachedDfSource` into `ContextIndexSearcher`?